### ISSUE TITLE
Style/StringLiterals: `link_to '#foo'` not detected as violating `double_quotes`

### ISSUE THREAD

**Issue by paulkogel** [bbatsov/rubocop#3037]
`link_to '#foo'` is not detected as violating `Style/StringLiterals` set to `double_quotes`.

Sample code with comments:
```ruby
link_to '#foo' # should raise a Rubocop error but doesn't :(
link_to 'jim', key: 'value' # correctly detected ^_^
link_to "bob", key: "other value" # passes, move along :)
```

## Expected behavior
`link_to '#foo'` should be detected as not using double quotes

## Actual behavior
it is not detected although it uses single quotes. When chaning the string to `'foo'` it does get detected.
Output from sample repository:

```bash
rubocop
Inspecting 2 files
.C

Offenses:

test.rb:2:9: C: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
link_to 'jim', key: 'value' # correctly detected ^_^
        ^^^^^
test.rb:2:21: C: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
link_to 'jim', key: 'value' # correctly detected ^_^
                    ^^^^^^^

2 files inspected, 2 offenses detected
```

## Steps to reproduce the problem
Please see sample repository: https://github.com/paulkogel/rubocop-bug
`bundle install` and run `rubocop`

## RuboCop version
0.39.0

```
$ rubocop -V
0.39.0 (using Parser 2.3.0.7, running on ruby 2.3.0 x86_64-darwin15)
```

**Comment by deivid-rodriguez** [comment]
Dupe of #3017, I think.

**Comment by jonas054** [comment]
Not exactly. We'll keep both issues open, because `#` is a 7-bit ASCII character, so it's a different bug than the failure to report strings with, for example, `ñ` in them.
